### PR TITLE
Aso Bid Adapter: add new bid adapter 

### PR DESCRIPTION
**Pull request by adserver-online**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

- test parameters for validating bids
```js
//banner
{
  bidder: 'aso',
  params: {
    zone: 73815
  }
}
//video
{
  bidder: 'aso',
  params: {
    zone: 34668
  }
}
```

- [x] Hello world checked

- contact email of the adapter’s maintainer: alex@adsrv.org
- [x] official adapter submission

- Docs PR https://github.com/prebid/prebid.github.io/pull/3046

### CODE REVIEW

**Review by osazos — CHANGES_REQUESTED** [review]
Hi @adserver-online,

A question about your `buildRequests()` method, If the adUnit is configured with both Banner and Video mediaTypes, you will bid on Banner only. Is it the expected behavior?

Here the missing part to validate the adapter:

- FloorPrices module support is required with Prebid.js 5.x.
- you have to provide a renderer for outsream video context.

File: modules/asoBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+      h: window.innerHeight,
+      w: window.innerWidth,
+    },
+    imp: [],
+    ext: {},
+    user: {}
+  };
+
+  if (bidRequest.params.attr) {
+    utils.deepSetValue(payload, 'site.ext.attr', bidRequest.params.attr);
```
[the remark this answers is not in the document]
Comment (adserver-online) [reply modules/asoBidAdapter.js:318]: Hi @osazos,

Thank you for your review.

Our backend does not support multiple media types in the same request. So yes it's expected behavior. We added a warning for now and we will improve in the future.

We added support for FloorPrices module and outstream renderer.

As for undocumented parameter `params.attr`. Its purpose is custom targeting (e.g. keywords, tags, etc). It is a rarely used feature that's why it was not documented. We've added a description to the bidder's md-file.

**Comment by osazos** [comment]
Hi @adserver-online, thanks for your quick reply. All seems ok for me.

**Review by osazos — APPROVED** [review]

**Comment by adserver-online** [comment]
@osazos We're sorry for the changes after approval. We've made outstream renderer better configurable. Please have a look.

**Review by osazos — APPROVED** [review]
LGTM

**Comment by osazos** [comment]
Hi @adserver-online,

I merged your PR, but we get a CI error due to the usage of `Object.entries`, see bellow.
I don't why this error has not been reported before as your tests were passed…

Can you make a new PR to quickly fix it? Ping me in comment when it's done.

```
Edge 17.17134 (Windows 10): Executed 6794 of 6801 (skipped 11) SUCCESS (32.632 secs / 18.375 secs)
IE 11.0 (Windows 10): Executed 6794 of 6801 (2 FAILED) (skipped 11) (34.419 secs / 22.694 secs)
Chrome 79.0.3945.79 (Windows 10): Executed 6794 of 6801 (skipped 11) SUCCESS (29.922 secs / 15.418 secs)
Chrome 90.0.4430.72 (Windows 10): Executed 6794 of 6801 (skipped 11) SUCCESS (19.792 secs / 10.409 secs)
Edge 90.0.818.39 (Windows 10): Executed 6794 of 6801 (skipped 11) SUCCESS (25.3 secs / 13.331 secs)
Firefox 88.0 (Windows 10): Executed 6794 of 6801 (skipped 11) SUCCESS (22.494 secs / 12.761 secs)
Safari 14.1 (Mac OS 10.15.7): Executed 6794 of 6801 (skipped 11) SUCCESS (15.078 secs / 10.441 secs)
Firefox 72.0 (Windows 10): Executed 6794 of 6801 (skipped 11) SUCCESS (22.481 secs / 13.021 secs)
Safari 12.1.2 (Mac OS 10.14.6): Executed 6794 of 6801 (skipped 11) SUCCESS (11.446 secs / 7.97 secs)
TOTAL: 2 FAILED, 61144 SUCCESS


1) should send GDPR/USP consent data if it applies
     Adserver.Online bidding adapter GDPR/USP compliance
     TypeError: Object doesn't support property or method 'entries'
   at getConsentsIds (webpack:///modules/asoBidAdapter.js:289:3 <- test/test_index.js:119801:21)
   at createBasePayload (webpack:///modules/asoBidAdapter.js:326:5 <- test/test_index.js:119842:5)
   at Anonymous function (webpack:///modules/asoBidAdapter.js:26:7 <- test/test_index.js:119538:7)
   at _each (webpack:///src/utils.js:399:29 <- test/test_index.js:603:29)
   at buildRequests (webpack:///modules/asoBidAdapter.js:25:5 <- test/test_index.js:119537:5)
   at Anonymous function (webpack:///test/spec/modules/asoBidAdapter_spec.js:194:7 <- test/test_index.js:119375:7)

2) should return iframe sync option
     Adserver.Online bidding adapter response handler getUserSyncs
     TypeError: Object doesn't support property or method 'entries'
   at getConsentsIds (webpack:///modules/asoBidAdapter.js:289:3 <- test/test_index.js:119801:21)
   at getUserSyncs (webpack:///modules/asoBidAdapter.js:117:9 <- test/test_index.js:119627:9)
   at Anonymous function (webpack:///test/spec/modules/asoBidAdapter_spec.js:333:9 <- test/test_index.js:119488:9)
```

**Comment by adserver-online** [comment]
Hi, 
We will fix it today asap

**Comment by ChrisHuie** [comment]
Merged the fix for ie 🙏 Thanks @adserver-online for jumping on this

**Pull request merged**